scenario/constraints/MigrationConstraint.ts

```diff
@@ -77,9 +79,13 @@ export class VerifyMigrationConstraint<T extends CometContext> implements Static
       async function (ctx: T): Promise<T> {
         const govDeploymentManager = ctx.world.auxiliaryDeploymentManager || ctx.world.deploymentManager;
         if (ctx.migrations) {
-          for (const migration of ctx.migrations) {
+          for (const migrationData of ctx.migrations) {
+            const migration = migrationData.migration;
             if (migration.actions.verify && !(await isEnacted(migration.actions, ctx.world.deploymentManager, govDeploymentManager))) {
-              await migration.actions.verify(ctx.world.deploymentManager, govDeploymentManager);
+              // XXX we really need to be capturing the block at which the proposal was executed instead of the migration being run
```

Comment:
This is the tricky part. We actually want to be capturing the block number after the proposal has been executed. Otherwise, the diff might also include state changes from other proposals, not related to this migration, being executed.

Reply:
Can we just pass it from ProposalConstraint?

Reply:
Although actually we should be verifying as of whatever block it is. Should we only be passing the pre block number?

Reply:
> Can we just pass it from ProposalConstraint?

Yeah I thought about passing it in from the ProposalConstraint, but we actually don't know what the proposal is for a specific migration unless we output that information from the migration itself.

> Although actually we should be verifying as of whatever block it is. Should we only be passing the pre block number?

Originally that's what I did. But I realized that we should probably be diffing the states of each migration in isolation. Reason being that there could be other proposals that get executed by the ProposalConstraint that would also alter the state, which would cause the `verify()` step for this migration to fail.

Reply:
But in that case, I think you want to run the whole verify check on the post block? You could make the same argument for all the checks it does

Reply:
Agreed, currently exploring this.

Reply:
I've implemented what we discussed offline. Basically, got rid of the `VerifyMigrationConstraint` and moved the verify step into the `ProposalConstraint`. This allows us to verify each migration right after its proposal is executed. We also keep track of which migrations have been verified already to make sure all migrations are verified, even those that are not tied to a specific proposal.